sim/mage/items.go

```diff
@@ -28,33 +30,61 @@ var ItemSetFirelordsVestments = core.NewItemSet(core.ItemSet{
 })
 
 // T12
-/* var ItemSetFirehawkRobesOfConflagration = core.NewItemSet(core.ItemSet{
-	Name:            "Firehawk Robes of Conflagration",
-	AlternativeName: "Firehawk",
+var ItemSetFirehawkRobesOfConflagration = core.NewItemSet(core.ItemSet{
+	Name: "Firehawk Robes of Conflagration",
 	Bonuses: map[int32]core.ApplyEffect{
+		// You have a chance to summon a Mirror Image to assist you in battle for 15 sec when you cast Frostbolt, Fireball, Frostfire Bolt, or Arcane Blast.
+		// (Proc chance: 20%, 45s cooldown)
 		2: func(agent core.Agent) {
-			character := agent.GetCharacter()
+			mage := agent.(MageAgent).GetMage()
 
-			core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
-				Name:           "Firehawk Mirror Image",
+			core.MakeProcTriggerAura(&mage.Unit, core.ProcTrigger{
+				Name:           "Item - Mage T12 2P Bonus",
 				Callback:       core.CallbackOnCastComplete,
 				ClassSpellMask: MageSpellArcaneBlast | MageSpellFireball | MageSpellFrostfireBolt | MageSpellFrostbolt,
-				ProcChance:     0.1, //Just a made up number for now
+				ProcChance:     0.20,
+				ICD:            time.Second * 45,
 				Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
-					&character.EnableWithTimeout(character.NewMirrorImage())
+					mage.t12MirrorImage.EnableWithTimeout(sim, mage.t12MirrorImage, time.Second*15)
 				},
 			})
 		},
+		// Your spells have an increased chance to trigger Brain Freeze or Hot Streak.
+		// In addition, Arcane Power decreases the cost of your damaging spells by 10% instead of increasing their cost.
 		4: func(agent core.Agent) {
-			character := agent.GetCharacter()
+			mage := agent.(MageAgent).GetMage()
 
-			character.AddStaticMod(core.SpellModConfig{
-				Kind:       core.SpellMod_CastTime_Pct,
-				ClassMask:  MageSpellArcaneBlast | MageSpellFireball | MageSpellFrostfireBolt | MageSpellFrostbolt,
-				FloatValue: -0.1,
-			})
+			core.MakePermanent(mage.RegisterAura(core.Aura{
+				Label: "Item - Mage T12 4P Bonus",
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					mage.brainFreezeProcChance += .15
+					mage.hotStreakProcChance += 0.30
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					mage.brainFreezeProcChance -= .15
+					mage.hotStreakProcChance -= .30
+				},
+			}))
+
+			if mage.arcanePowerCostMod != nil {
```

Comment:
Did you debug / verify this? I think I had to do a callback here because ItemEffects were initialized before the talents. Only the agent constructor is called before.

```go
mage.OnSpellRegistered(func(spell *core.Spell) {
  if (spell.ClassSpellMask != MageSpellArcanePower) return;

  // do stuff
}
```

BTW while checking this I noticed that the ArcanePower spell doesn't even define a class mask, thus, the `ArcaneFlows` talent isn't applied to it properly :D

Reply:
Oh I forgot to check that thank you. And oof yeah it looks like there's a class mask defined but not applied to the AP spell